quipucords/tests/api/reports/test_reports.py

```diff
@@ -168,13 +168,22 @@ def test_report_tarball_details_json(
     )
     assert api_details_response.ok, api_details_response.text
     api_details_json = api_details_response.json()
+
+    # TODO FIXME What is causing updated_at to be different in these responses?
+    # Upon initial investigation, the created_at times in the responses match,
+    # but api_details_json's updated_at is more recent than tarball_details_json's.
+    # The difference is on the order of microseconds which means me to believe that
+    # something is triggering an unexpected save between the first and second get.
+    del tarball_details_json["updated_at"]
+    del api_details_json["updated_at"]
```

Comment:
the csv generation is the root cause here. If the csv is not stored in the model, it will be generated on the fly.

when we ask for the details report, we ask only for json, so no csv is created (and even if we asked, there's a "bug" here, because the datetimes will be serialized before the model is updated).

finally, when we ask for the tarball, the csv will be created and the model will be updated.
or at least this is the most likely explanation that comes up here.